editor/Scripting/ScriptManager.cs

```diff
@@ -111,6 +114,19 @@ public IEnumerable<string> GetScriptNames()
             }
         }
 
+        public void UpdateReferencedAssemblies(string[] referencedAssemblies)
+        {
+            this.referencedAssemblies = referencedAssemblies;
+
+            foreach(var entry in scriptContainers)
+            {
+                var scriptContainer = (ScriptContainerAppDomain<TScript>)entry.Value;
+                scriptContainer.ReferencedAssemblies = referencedAssemblies;
```

Comment:
You shouldn't cast here, make ReferencedAssemblies accessible from the base class.